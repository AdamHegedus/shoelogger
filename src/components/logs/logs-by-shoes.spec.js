import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import BootstrapVue from 'bootstrap-vue';
import LogsByShoes from '@/components/logs/logs-by-shoes';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(BootstrapVue);

describe('logs-by-shoes.vue', () => {
    describe('on init', () => {
        it('should render component', () => {
            // ASSIGN
            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Test',
                        brand: 'Foo',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const shoesState = {
                shoes: [
                    {
                        id: 0,
                        brand: 'Foo',
                        product: 'Bar'
                    },
                    {
                        id: 1,
                        brand: 'Foo',
                        product: 'Test'
                    }
                ]
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions,
                        state: shoesState
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            // ACT
            const wrapper = shallowMount(LogsByShoes, { store, localVue });

            // ASSERT
            expect(wrapper.vm.selectedShoe).toMatchSnapshot();
            expect(wrapper.element).toMatchSnapshot();
            expect(navigationActions.setNavigation.mock.calls[0][1]).toMatchSnapshot();
            expect(shoesActions.getShoes).toBeCalled();
        });
    });

    describe('getShoeDisplayValue', () => {
        it('should set selected shoe', () => {
            // ASSIGN
            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Test',
                        brand: 'Foo',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const shoesState = {
                shoes: [
                    {
                        id: 0,
                        brand: 'Foo',
                        product: 'Bar'
                    },
                    {
                        id: 1,
                        brand: 'Foo',
                        product: 'Test'
                    }
                ]
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions,
                        state: shoesState
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(LogsByShoes, { store, localVue });
            const input = {
                id: 0,
                brand: 'Foo',
                product: 'Bar'
            };

            // ACT
            const actual = wrapper.vm.getShoeDisplayValue(input);

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(wrapper.element).toMatchSnapshot();
        });
    });

    describe('getLogsByProduct', () => {
        it('should return filtered list', () => {
            // ASSIGN
            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Test',
                        brand: 'Foo',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const shoesState = {
                shoes: [
                    {
                        id: 0,
                        brand: 'Foo',
                        product: 'Bar'
                    },
                    {
                        id: 1,
                        brand: 'Foo',
                        product: 'Test'
                    }
                ]
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions,
                        state: shoesState
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(LogsByShoes, { store, localVue });
            const product = 'Bar';

            // ACT
            const actual = wrapper.vm.getLogsByProduct(product);

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });

    describe('deleteLog', () => {
        it('should call action', () => {
            // ASSIGN
            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Test',
                        brand: 'Foo',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const shoesState = {
                shoes: [
                    {
                        id: 0,
                        brand: 'Foo',
                        product: 'Bar'
                    },
                    {
                        id: 1,
                        brand: 'Foo',
                        product: 'Test'
                    }
                ]
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions,
                        state: shoesState
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(LogsByShoes, { store, localVue });
            const input = 0;

            // ACT
            wrapper.vm.deleteLog(input);

            // ASSERT
            expect(logsActions.deleteLog.mock.calls[0][1]).toMatchSnapshot();
        });
    });

    describe('columnDefinitions', () => {
        it('should return correct value', () => {
            // ASSIGN
            const logsActions = {
                getLogs: jest.fn(),
                deleteLog: jest.fn(),
                resetMeta: jest.fn()
            };

            const logsState = {
                logs: [
                    {
                        id: 0,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 1,
                        product: 'Bar',
                        brand: 'Foo',
                        distance: 1,
                        timestamp: 100
                    },
                    {
                        id: 2,
                        product: 'Test',
                        brand: 'Foo',
                        distance: 3,
                        timestamp: 100
                    }
                ]
            };

            const shoesActions = {
                getShoes: jest.fn()
            };

            const shoesState = {
                shoes: [
                    {
                        id: 0,
                        brand: 'Foo',
                        product: 'Bar'
                    },
                    {
                        id: 1,
                        brand: 'Foo',
                        product: 'Test'
                    }
                ]
            };

            const navigationActions = {
                setNavigation: jest.fn()
            };

            const store = new Vuex.Store({
                modules: {
                    logs: {
                        namespaced: true,
                        actions: logsActions,
                        state: logsState
                    },
                    shoes: {
                        namespaced: true,
                        actions: shoesActions,
                        state: shoesState
                    },
                    navigation: {
                        namespaced: true,
                        actions: navigationActions
                    }
                }
            });

            const wrapper = shallowMount(LogsByShoes, { store, localVue });

            // ACT
            const actual = wrapper.vm.columnDefinitions;

            // ASSERT
            expect(actual).toMatchSnapshot();
        });
    });
});
