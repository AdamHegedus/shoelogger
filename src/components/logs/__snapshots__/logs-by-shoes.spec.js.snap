// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logs-by-shoes.vue deleteLog should call action 1`] = `
Object {
  "id": 0,
}
`;

exports[`logs-by-shoes.vue getLogsByProduct should return filtered list 1`] = `
Array [
  Object {
    "brand": "Foo",
    "distance": 1,
    "id": 0,
    "product": "Bar",
    "timestamp": 100,
  },
  Object {
    "brand": "Foo",
    "distance": 1,
    "id": 1,
    "product": "Bar",
    "timestamp": 100,
  },
]
`;

exports[`logs-by-shoes.vue on init should render component 1`] = `null`;

exports[`logs-by-shoes.vue on init should render component 2`] = `
<div
  class="logs-by-shoes"
>
  <h3>
    Logs by Shoes
  </h3>
   
  <div
    class="col-12 col-lg-6 col-lg-6 px-0 py-2"
  >
    <b-dropdown-stub
      boundary="scrollParent"
      class="btn-block"
      id="type"
      no-flip="true"
      role="menu"
      size="lg"
      text="Select Shoe"
      toggleclass="btn btn-secondary btn-lg btn-block shadow"
      toggletext="Toggle Dropdown"
    >
      <b-dropdown-item-stub
        activeclass="active"
        class="[object Object]"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
      >
        
                Foo - Bar
            
      </b-dropdown-item-stub>
      <b-dropdown-item-stub
        activeclass="active"
        class="[object Object]"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
      >
        
                Foo - Test
            
      </b-dropdown-item-stub>
    </b-dropdown-stub>
  </div>
   
  <div
    class="col-12 col-lg-6 col-lg-6 px-0 py-2"
  >
    <!---->
  </div>
</div>
`;

exports[`logs-by-shoes.vue on init should render component 3`] = `"/logs/by-shoes"`;

exports[`logs-by-shoes.vue selectShoe should set selected shoe 1`] = `
Object {
  "brand": "Foo",
  "id": 0,
  "product": "Bar",
}
`;

exports[`logs-by-shoes.vue selectShoe should set selected shoe 2`] = `
<div
  class="logs-by-shoes"
>
  <h3>
    Logs by Shoes
  </h3>
   
  <div
    class="col-12 col-lg-6 col-lg-6 px-0 py-2"
  >
    <b-dropdown-stub
      boundary="scrollParent"
      class="btn-block"
      id="type"
      no-flip="true"
      role="menu"
      size="lg"
      text="Foo Bar"
      toggleclass="btn btn-secondary btn-lg btn-block shadow"
      toggletext="Toggle Dropdown"
    >
      <b-dropdown-item-stub
        activeclass="active"
        class="[object Object]"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
      >
        
                Foo - Bar
            
      </b-dropdown-item-stub>
      <b-dropdown-item-stub
        activeclass="active"
        class="[object Object]"
        event="click"
        exactactiveclass="active"
        routertag="a"
        target="_self"
      >
        
                Foo - Test
            
      </b-dropdown-item-stub>
    </b-dropdown-stub>
  </div>
   
  <div
    class="col-12 col-lg-6 col-lg-6 px-0 py-2"
  >
    <b-table-stub
      apiurl=""
      bordered="true"
      currentpage="1"
      emptyfilteredtext="There are no records matching your request"
      emptytext="There are no records to show"
      fields="[object Object],[object Object],[object Object]"
      footvariant=""
      headvariant=""
      hover="true"
      items="[object Object],[object Object]"
      labelsortasc="Click to sort Ascending"
      labelsortdesc="Click to sort Descending"
      perpage="0"
      sortdirection="asc"
      value=""
    />
  </div>
</div>
`;
